las_geoh5/import_las.py

```diff
@@ -373,23 +373,42 @@ def las_to_drillhole(
 
     if not isinstance(data, list):
         data = [data]
-    if not isinstance(survey, list):
-        survey = [survey] if survey else []
+    if not isinstance(surveys, list):
+        surveys = [surveys] if surveys else []
 
     for datum in tqdm(data, desc="Adding drillholes and data to workspace"):
         collar = get_collar(datum, translator, logger)
         if all(k == 0 for k in collar) and options.skip_empty_header:
             continue
 
-        drillhole = create_or_append_drillhole(
+        create_or_append_drillhole(
             datum,
             drillhole_group,
             property_group,
             translator=translator,
             logger=logger,
             collocation_tolerance=options.collocation_tolerance,
         )
-        ind = [drillhole.name == s.name.rstrip(".las") for s in survey]
-        if any(ind):
-            survey_path = survey[np.where(ind)[0][0]]
-            _ = add_survey(survey_path, drillhole, logger)
+
+    for drillhole in tqdm(drillhole_group.children, desc=""):

```

Comment:
This seems like it should all go inside add_survey.  Pass the whole list of surveys into it and resolve the from file vs. max depth inside?

Reply:
The check for single line survey becomes more complicated as the same drillhole could be hit multiple times depending on the file structure (duplicates). In which case i would need to do some comparison. It's another loop, but the logic for survey would only get hit once per hole here.